### PR TITLE
[MM-58840] Add routing restrictions

### PR DESCRIPTION
**Pull request by JulienTant**
#### Summary
This PR adds some rules to ensure IDs respect a certain format in the system console.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-58840

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: e2e-tests/cypress/tests/integration/channels/system_console/user_management_spec.js [hunk 1]
```diff
@@ -222,6 +222,17 @@ describe('User Management', () => {
...
 
+    it("MM-58840 Users - can't navigate to invalid URL", () => {
+        // # Login as sysadmin.
+        cy.apiLogin(sysadmin);
+
+        // # Visit the invalid URL.
+        cy.visit('/admin_console/user_management/user/invalid');
+
+        // * Verify that the user is redirected to the default page.
+        cy.url().should('include', '/admin_console/about/license');
```
Comment (JulienTant) [inline e2e-tests/cypress/tests/integration/channels/system_console/user_management_spec.js:233]: this one is handled differently compared to the other ones use routes like `/groups` for listing  and `/groups/{id}` for detail, but user uses `/users` for listing but `/user/{id}` (no **S** after user) to show a user.

**Review by sbishel — APPROVED** [review]

**Review by enahum — APPROVED** [review]

**Review by lindy65 — APPROVED** [review]
Thanks @JulienTant - QA-deferring to be tested post merge

**Review by enzowritescode — APPROVED** [review]

**Pull request merged**